jenkins/bootstrap.py

```diff
@@ -222,13 +222,14 @@ def checkout(call, repo, branch, pull, ssh=False, git_cache=''):
         except OSError:
             pass
         call([git, 'init', repo, '--separate-git-dir=%s' % cache_dir])
-        # Remove any changes left around from previous runs.
-        call([git, 'clean', '-dfx'])
-        call([git, 'reset', '--hard'])
     else:
         call([git, 'init', repo])
     os.chdir(repo)
 
+    # Remove any changes left around from previous runs.
+    call([git, 'clean', '-dfx'])
```

Comment:
Please hide this behind a flag, and leave it disabled by default for human users.

Also will you add test coverage for this scenario since we've broken things twice?